analyzers/src/SonarAnalyzer.Common/Rules/PropertiesAccessCorrectFieldBase.cs

```diff
@@ -251,6 +260,13 @@ public IFieldSymbol GetSingleMatchingFieldOrNull(IPropertySymbol propertySymbol)
 
             private static bool AreCanonicalNamesEqual(string name1, string name2) =>
                 name1.Equals(name2, StringComparison.OrdinalIgnoreCase);
+
+            private bool FieldMatchesTheProperty(IFieldSymbol field, IPropertySymbol property) =>
+                // We're not caching the property name as only expect to be called once per property
+                AreCanonicalNamesEqual(fieldToStandardNameMap[field], GetCanonicalFieldName(property.Name))
```

Comment:
Nitpick: I would rename the GetCanonicalFieldName to GetCanonicalName as it does the same for property names too.